src/IO/S3/Client.cpp

```diff
@@ -934,6 +946,24 @@ void Client::BuildHttpRequest(const Aws::AmazonWebServiceRequest& request,
     }
 }
 
+std::string Client::getGCSOAuthToken() const
+{
+    if (provider_type != ProviderType::GCS)
+        return "";
+
+    const auto & http_client = GetHttpClient();
+
+    if (!http_client)
+        return "";
+
+    auto * gcp_oauth_client = dynamic_cast<PocoHTTPClientGCPOAuth *>(http_client.get());
+
+    if (!gcp_oauth_client)
+        return "";
```

Comment:
should these checks be `chassert` 🤔 
maybe safer to just return empty token as it's not critical function